lua/lspconfig/configs.lua

```diff
@@ -178,7 +181,7 @@ function configs.__newindex(t, config_name, config_def)
       end)
     end
 
-    -- Used by :LspInfo
+    -- Used by :LspInfo (evil, mutable aliases?)
     M.get_root_dir = get_root_dir
     M.filetypes = config.filetypes
     M.handlers = config.handlers
```

Comment:
It's not clear why these fields don't merge the info from `config_def`. @glepnir do you know?

Reply:
should be merged with user config.  looks like a mistake here.  but the handlers are no longer managed by lspconfig. So it is useless.

Reply:
> but the handlers are no longer managed by lspconfig. So it is useless.

Interesting, can some code be deleted?

Reply:
see https://github.com/neovim/nvim-lspconfig/blob/master/lua/lspconfig/configs/denols.lua

we suggest use vim.lsp.handlers[method] = xxx, but lspinfo can not capture this override.

Reply:
If we want to show the override handler, I think we can locate the modified function file through debug.info in checkhealth? then i think we can remove handlers manage here.

Reply:
I'm trying to understand:

> handlers are no longer managed by lspconfig

in the denols.lua config you linked, I still see `config.handlers` being assigned.

Reply:
More like a bridge. It passes to vim.lsp.handlers via config. lspconfig should not do this. It is better to use vim.lsp.handlers directly in config.

there only `jdtls eslint denols relay_lsp koka grammarly` use config.handlers.

Reply:
could you create an issue (or PR to lspconfig.txt) that describes how users / config authors can do that to migrate their `lspconfig.handlers` ?

Reply:
```diff
diff --git a/runtime/lua/vim/lsp/health.lua b/runtime/lua/vim/lsp/health.lua
index 18066a84d..faf43ddcd 100644
--- a/runtime/lua/vim/lsp/health.lua
+++ b/runtime/lua/vim/lsp/health.lua
@@ -159,6 +159,16 @@ local function check_position_encodings()
   end
 end

+function M.check_handlers_override()
+  local handlers = vim.lsp.handlers
+  for name, handler in ipairs(handlers) do
+    local source = debug.getinfo(handler, 'S').source
+    if not source:find('vim/lsp/handlers.lua') then
+      report_info(('Handler of method %s override in: %s '):format(name, source))
+    end
+  end
+end
+
 --- Performs a healthcheck for LSP
 function M.check()
   check_log()

```

an example which improve our checkhealth of vim.lsp to show the override handlers location. then we don't need to show it in lspinfo .this way is more like we want.

Reply:
`just `vim.lsp.handler[method]` replace `config_def.handlers[method]`  yes ..for more clear we need an issue or Pr for track..

Reply:
> for more clear we need an issue or Pr for track

yes,don't want to forget this thread.